### PR TITLE
mysql-compatibility: update the doc for v5.1

### PR DESCRIPTION
**Pull request by bb7133**
### What is changed, added or deleted? (Required)

Update 'MySQL compatibility' docs because we'll support most 'changing column types' DDL statements.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?
- NA

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: mysql-compatibility.md [hunk 1]
```diff
@@ -95,13 +95,10 @@ See also: [TiDB SQL Grammar](https://pingcap.github.io/sqlgram/#functioncallkeyw
...
-* Adding/Dropping the primary key of the `CLUSTERED` type is unsupported. For more details about the primary key of the `CLUSTERED` type, refer to [clustered index](/clustered-indexes.md).
-* Changing the field type to its superset is unsupported. For example, TiDB does not support changing the field type from `INTEGER` to `VARCHAR`, or from `TIMESTAMP` to `DATETIME`. Otherwise, the error information `Unsupported modify column: type %d not match origin %d` might be output.
-* Change/Modify data type does not currently support "lossy changes", such as changing from BIGINT to INT.
-* Change/Modify decimal columns does not support changing the precision.
-* Change/Modify integer columns does not permit changing the `UNSIGNED` attribute.
+* Some type changings through `ALTER TABLE` statement is not supported. For example, TiDB does not support changing the field type from `DECIMAL` to `DATE`. Otherwise, the error information `Unsupported modify column: type %d not match origin %d` might be output. See [`ALTER TABLE`](/sql-statements/sql-statement-alter-table.md) for further details.
```
Comment (morgo) [inline mysql-compatibility.md:98]:
```suggestion
* Some type changings through `ALTER TABLE` statement are not supported. For example, TiDB does not support changing the field type from `DECIMAL` to `DATE`. When a type change is not supported, the error  `Unsupported modify column: type %d not match origin %d` will be returned. See [`ALTER TABLE`](/sql-statements/sql-statement-alter-table.md) for further details.
```

**Review by zimulala — COMMENTED** [review]
Update `sql-statement-modify-column.md` too?

**Comment by AilinKid** [comment]
> Update `sql-statement-modify-column.md` too?

https://github.com/pingcap/docs-cn/pull/6413

File: mysql-compatibility.md (hunk 1 above)
Comment (bb7133) [reply mysql-compatibility.md:98]: Thanks, for your suggestion, it has been addressed in @AilinKid 's https://github.com/pingcap/docs-cn/pull/6413.

**Comment by bb7133** [comment]
I'll close this PR since it has been covered by pingcap/docs-cn#6413

@TomShawn @AilinKid @morgo @zimulala Thanks for your reviews.

**Pull request closed without merging**